### ISSUE TITLE
threading.WaitForMultipleObjectsEx.test2 failed on CentOS7.1

### ISSUE THREAD

**Issue by wtgodbe** [dotnet/runtime#6327]
http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/checked_centos7.1/1435/testReport/threading/WaitForMultipleObjectsEx/test2/

This happened on July 9 in a test job on merged code. Worth some investigation (FWIW, I'm only seeing this 1 failure for this test).

**Comment by v-haren** [comment]
failed again in job: [runtime-coreclr ilasm 20200919.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=822524&view=ms.vss-test-web.build-test-results-tab&runId=26184806&resultId=107664&paneView=debug)

failed test: threading.WaitForMultipleObjectsEx.test2

CoreCLR Linux arm64 Checked ilasmroundtrip @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-a45aeeb-20190620155855

Error message
~~~
Expected thread to wait for 1000 ms (and get interrupted).
Interrupt Time: 500 ms,  ThreadWaitDelta 499


Stack trace
~~~

**Comment by v-haren** [comment]
failed again in job: [runtime-coreclr jitstress2-jitstressregs 20201004.1](https://dev.azure.com/dnceng/public/_build/results?buildId=841296&view=ms.vss-test-web.build-test-results-tab&runId=26798506&paneView=debug&resultId=108473)

failed test: threading.WaitForMultipleObjectsEx.test2

CoreCLR Linux arm64 Checked jitstress2_jitstressregs3 @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-a45aeeb-20190620155855

Error message
~~~
Expected thread to wait for 1000 ms (and get interrupted).
Interrupt Time: 500 ms,  ThreadWaitDelta 499

Stack trace
~~~

**Comment by janvorli** [comment]
Actually, I've checked the types of the three variables and they are all unsigned! So 499-500 is not -1, but 0xffffffff, which is obviously > 10. That explains why the test goes the failure path. 
And I've also realized that the second part of the condition is wrong, it first tests if the ThreadWaitDelta_WFMO_test2 is less than the InterruptTime and then checks if the ThreadWaitDelta_WFMO_test2 - InterruptTime, which would always be <= 0, is larger than the TOLERANCE. 
So it is a test bug. The subtraction should be reversed. The test should pass as the 1ms delta is within the tolerance.
